src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -244,18 +230,33 @@ private boolean executeCmd(List<String> cmds) throws IOException, InterruptedExc
         }
 
         private Result executeCmd(String cmd) throws IOException, InterruptedException {
-            PrintStream stdout = listener.getLogger();
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            TeeOutputStream stdout = new TeeOutputStream(listener.getLogger(), baos);
             PrintStream stderr = listener.getLogger();
 
-            boolean result = launcher.launch()
-                    .envs(build.getEnvironment(listener))
-                    .pwd(build.getWorkspace())
-                    .stdout(stdout)
-                    .stderr(stderr)
-                    .cmdAsSingleString(cmd)
-                    .start().join() == 0;
-            String stdoutStr = stdout.toString();
-            return new Result(result, stdoutStr);
+            KeyMaterial key = null;
+            try {
+                // get Docker registry credentials
+                key = registry.materialize(build);
+
+                EnvVars env = new EnvVars();
+                env.putAll(build.getEnvironment(listener));
+                env.putAll(key.env());
+    
+                boolean result = launcher.launch()
+                        .envs(env)
+                        .pwd(build.getWorkspace())
+                        .stdout(stdout)
+                        .stderr(stderr)
+                        .cmdAsSingleString(cmd)
+                        .start().join() == 0;
+                return new Result(result, baos.toString("UTF8"));
```

Comment:
```UTF-8```?
But I'm unsure if this will work on platforms that doesn't use UTF-8 as platform encoding.